spec/calculator.spec.js

```diff
@@ -16,288 +16,304 @@
-      instance.addNumber(3);
+      instance.addNumber('2');
       instance.addMinus();
-      expect(instance.input()).to.equal(3);
-
-      instance.clear();
+      expect(instance.input()).to.equal("6");
+    });
 
-      instance.addNumber(2);
+    it('It should perform the result of the old multiply operation before moving to new operation if there was more than one operation', () => {
+      instance.addNumber('6');
       instance.addMultiplication();
-      instance.addNumber(6);
+      instance.addNumber('2');
       instance.addMultiplication();
-      expect(instance.input()).to.equal(12);
-      instance.addNumber(2);
-      instance.equal();
-      expect(instance.input()).to.equal(24);
-
-      instance.clear();
+      expect(instance.input()).to.equal("12");
+    });
 
-      instance.addNumber(1);
-      instance.addNumber(6);
+    it('It should perform the result of the old divide operation before moving to new operation if there was more than one operation', () => {
+      instance.addNumber('9');
       instance.addDivision();
-      instance.addNumber(2);
+      instance.addNumber('3');
       instance.addDivision();
-      expect(instance.input()).to.equal(8);
-      instance.addNumber(2);
-      instance.equal();
-      expect(instance.input()).to.equal(4);
-      
-      instance.clear();
-
-      instance.addNumber(1);
-      instance.addPlus();
-      instance.addNumber(2);
-      instance.addMinus();
-      instance.addNumber(1);
-      instance.equal();
-      expect(instance.input()).to.equal(2);
-
-      instance.clear();
-
-      instance.addNumber(2);
-      instance.addMultiplication();
-      instance.addNumber(5);
-      instance.addPlus();
-      expect(instance.input()).to.equal(10);
-      instance.addNumber(2);
-      expect(instance.input()).to.equal("2");
-      instance.addMinus();
-      expect(instance.input()).to.equal(12);
-      instance.addNumber(3);
       expect(instance.input()).to.equal("3");
-      instance.addMultiplication();
-      expect(instance.input()).to.equal(9);
-      instance.addNumber(3);
-      instance.addMinus();
-      expect(instance.input()).to.equal(27);
-      instance.addNumber(1);
-      instance.addDivision();
-      expect(instance.input()).to.equal(26);
-      instance.addNumber(2);
-      instance.equal();
-      expect(instance.input()).to.equal(13);
-
     });
 
     it('It should perform the last operation', () => {
-      instance.addNumber(3);
+      instance.addNumber('3');
       instance.addPlus();
       instance.addMinus();
-      instance.addNumber(1);
+      instance.addNumber('1');
       instance.equal();
-      expect(instance.input()).to.equal(2);
+      expect(instance.input()).to.equal("2");
     });
 
-    it('It should show correct result even if user pressed an operation button twice', () => {
-      instance.addNumber(5);
+    it('It should show correct result even if user pressed an subtract operation button twice', () => {
```

Comment:
Maybe it would be better to rename these 4 tests to something like "It should ignore consecutive presses of the X operator"